packages/components/src/components/action-bar/action-bar.tsx

```diff
@@ -68,8 +71,33 @@ export class ActionBar extends LitElement {
     const groupCount =
       this.hasActionsEnd || !expandDisabled ? actionGroups.length + 1 : actionGroups.length;
 
+    let bufferSize = groupCount;
+    const actionBarContainerStyle = getComputedStyle(this.containerRef.value);
+
+    if (layout === "horizontal") {
+      bufferSize +=
+        parseInt(actionBarContainerStyle.paddingInlineStart) +
+        parseInt(actionBarContainerStyle.paddingInlineEnd);
+
+      if (this.actionGroups.length > 0) {
+        this.actionGroups.forEach((actionGroup, i) => {
```

Comment:
`this.actionGroups` is referenced a few times, so it could be stored in a local var.